libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1024,7 +1051,7 @@ conformsToSpecM ::
   forall fn a m. (HasSpec fn a, MonadGenError m) => a -> Specification fn a -> m ()
 conformsToSpecM _ TrueSpec = pure ()
 conformsToSpecM a (MemberSpec as) =
-  if elem a as
+  if elem a (NE.toList as)
```

Comment:
`elem` is polymorphic enough, you don't need `NE.toList as` here AFAIK

Reply:
Yeah, I never thought about that. All those foldable methods will work on NE.NonEmpty as well as Lists etc.